src/navigation-plan.js

```diff
@@ -11,6 +12,15 @@ export function buildNavigationPlan(navigationContext, forceLifecycleMinimum) {
   var next = navigationContext.nextInstruction;
   var plan = {}, viewPortName;
 
+  if ('redirect' in next.config) {
+    let redirectLocation = resolveUrl(next.config.redirect, getInstructionBaseUrl(next));
```

Comment:
We need to resolve the `next.config.redirect` relative to the router it was registered in, but the router's `baseUrl` isn't set until a navigation is committed in it. Instead we walk up the chain of current instructions and build up the baseUrl to use for the redirect.